### ISSUE TITLE
AmbiguousBlockAssociation False Positive

### ISSUE THREAD

**Issue by ashmaroli** [bbatsov/rubocop#4227]
`Lint/AmbiguousBlockAssociation` cop raises a false alarm against non-dubious block association.

## Example Code

```ruby
assert_equal site.posts.find { |p| p.title == "Foo Bar" }, results.first
```

## Expected behavior

Does not expect Rubocop to flag the code above and even if it did, I expect the message to tell me to parenthesize the param to associate the block with `find` call.

## Actual behavior

```
W: Lint/AmbiguousBlockAssociation: Parenthesize the param find to make sure that the block will
be associated with the assert_equal method call.
        assert_equal site.posts.find { |p| p.title == "Foo Bar" }, results.first
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Run `rubocop` against following:
```ruby
assert_equal site.posts.find { |p| p.title == "Foo Bar" }, results.first
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.3 i386-mingw32)
```

**(Also tested with latest `master` branch)** 
(similar issue: [Comment](https://github.com/bbatsov/rubocop/issues/4181#issuecomment-289512746) by @stephengroat)

/cc @smakagon

**Comment by smakagon** [comment]
@ashmaroli thanks for reporting. Created Pull Request that fixes this case.
